sock-merchant.ts

```diff
@@ -0,0 +1,18 @@
+interface Color {
+  [key: number]: number;
+}
+
+export function sockMerchant(ar: number[]): number {
+  const colorsCount = ar.reduce((allColors: Color, color) => {
+    if (allColors[color]) {
+      allColors[color]++;
+      return allColors;
+    }
+    allColors[color] = 1;
+    return allColors;
+  }, {});
+
+  return Object.values(colorsCount).reduce((count, value) => {
+    return count + Math.floor(value / 2);
+  }, 0);
+}
```

Comment:
you can reduce code boilerplate code if you have a single line in your function body

```suggestion
  return Object.values(colorsCount).reduce((count, value) => count + Math.floor(value / 2), 0);
}
```